lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -1087,10 +1087,10 @@ def ensure(self):
                 self.module.exit_json(**response)
             else:
                 response['changed'] = True
+                if failure_response['failures']:
```

Comment:
I think this should be below the following `if self.module.check_mode` conditional so that `check_mode` takes precedent.

Reply:
If the failure block is after the check block, then it needs to be duplicated inside the check block as well. Otherwise the module will not report errors during check mode.

As the PR currently stands, the module will not make any changes if even one of the items it tries to install will fail, so the guarantee of check mode not changing anything is still maintained by having the failure block before the check mode block.

Should I duplicate the code, or add a comment pointing out the failure conditional still guarantees check mode? There are benefits to both.

Reply:
Yeah, that's fair. I was thinking the `check_mode` reporting should take precedence but the error probably should. 👍